File: 卡码/代码随想录/C++/哈希/06-赎金信/LeetCode_0383.cpp
//383. Êê½ğĞÅ
#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
  bool canConstruct(string ransomNote, string magazine) {
		if(ransomNote.length()>magazine.length()) return false;
		int hash[26]={0};
		for(char ch:magazine)
			hash[ch-'a']++;
		for(char ch:ransomNote){
			hash[ch-'a']--;
			if(hash[ch-'a']<0) return false;
		}
		return true;
  }
};

File: 卡码/代码随想录/Python/哈希/06-赎金信/LeetCode_0383.py
#383. 赎金信
from typing import *

class Solution:
	def canConstruct(self, ransomNote: str, magazine: str) -> bool:
		if len(ransomNote)>len(magazine):
			return False
		hash=[0 for x in range(26)]
		for ch in magazine:
			hash[ord(ch)-ord('a')]+=1
		for ch in ransomNote:
			hash[ord(ch)-ord('a')]-=1
			if hash[ord(ch)-ord('a')]<0:
				return False
		return True